v0/destinations/pinterestConversion/utils.js

```diff
@@ -0,0 +1,248 @@
+/* eslint-disable no-param-reassign */
+const sha256 = require("sha256");
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  constructPayload,
+  isDefinedAndNotNull,
+  getHashFromArray
+} = require("../../util");
+const { COMMON_CONFIGS } = require("./config");
+
+const VALID_ACTION_SOURCES = ["app_android", "app_ios", "web", "offline"];
+
+const ecomEventMaps = [
+  {
+    src: ["order completed"],
+    dest: "Checkout"
+  },
+  {
+    src: ["product added"],
+    dest: "AddToCart"
+  },
+  {
+    src: ["products searched", "product list filtered"],
+    dest: "Search"
+  }
+];
+
+/**
+ *
+ * @param {*} userPayload Payload mapped from user fields
+ * @returns
+ * Further Processing the user fields following the instructions of Pinterest Conversion API
+ * Ref: https://s.pinimg.com/ct/docs/conversions_api/dist/v3.html
+ */
+const processUserPayload = userPayload => {
+  let formatValue;
+  Object.keys(userPayload).forEach(key => {
+    switch (key) {
+      case "em":
+        formatValue = userPayload[key].toString().toLowerCase();
+        userPayload[key] = [sha256(formatValue)];
+        break;
+      case "ph":
+      case "zp":
+        // zip fields should only contain digits
+        formatValue = userPayload[key].toString().replace(/[^0-9]/g, "");
+        if (key === "ph") {
+          // phone numbers should not contain leading zeros
+          formatValue = formatValue.replace(/^0+/, "");
+        }
+        userPayload[key] = [sha256(formatValue)];
+        break;
+      case "ge":
+      case "ct":
+      case "st":
+      case "country":
+        userPayload[key] = [sha256(formatValue)];
+        break;
+      case "db":
+      case "ln":
+      case "fn":
+      case "hashed_maids":
+        userPayload[key] = [userPayload[key]];
+        break;
+      default:
+        userPayload[key] = String(userPayload[key]);
+    }
+  });
+  return userPayload;
+};
```

Comment:
For some fields I am seeing there is the entry of `hashToSha256` in `pinterestUserConfig.json`, those will be doubleHashed here. And we should take permission from the customer if we have to hash the data or if the data coming to us is already hashed.

Reply:
only db, ln , fn and hashed_maids have been hashed inside the mapping JSON, hence, it is not being done here. Others are not. Rechecked and confirmed

Reply:
only db, ln , fn and hashed_maids have been hashed inside the mapping JSON, hence, it is not being done here. Others are not. Rechecked and confirmed

Reply:
Okay, what about checking if the data is already hashed or not?

Reply:
added this via integrations object